src/avram/database.cr

```diff
@@ -190,20 +199,19 @@ abstract class Avram::Database
     end
   end
 
-  private def transactions
-    self.class.transactions
+  private def connections
+    self.class.connections
   end
 
   private def wrap_in_transaction
-    db.transaction do |tx|
-      transactions[Fiber.current.object_id] ||= tx
+    (current_transaction || current_connection).transaction do
```

Comment:
Use the current transaction to make a sub transaction if one is available to use before using the connection